src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/SupplementaryEvidenceCreatorTest.java

```diff
@@ -37,26 +49,115 @@
     @Autowired
     private EnvelopeEventProcessor envelopeEventProcessor;
 
+    private static final String CREATE_EXCEPTION_RECORD_SUBMIT_URL = Environment.CASE_SUBMIT_URL
+        .replace(CASE_TYPE_BULK_SCAN, CASE_TYPE_EXCEPTION_RECORD);
+
+    private static final String CREATE_EXCEPTION_START_EVENT_URL = Environment.CASE_EVENT_TRIGGER_START_URL
+        .replace(CASE_TYPE_BULK_SCAN, CASE_TYPE_EXCEPTION_RECORD);
+
+    private static final String SERVICE_AUTHORIZATION_HEADER = "ServiceAuthorization";
+
+    // see WireMock mapping json files
+    private static final String MOCKED_IDAM_TOKEN_SIG = "q6hDG0Z1Qbinwtl8TgeDrAVV0LlCTRtbQqBYoMjd03k";
+    private static final String MOCKED_S2S_TOKEN_SIG =
+        "X1-LdZAd5YgGFP16-dQrpqEICqRmcu1zL_zeCLyUqMjb5DVx7xoU-r8yXHfgd4tmmjGqbsBz_kLqgu8yruSbtg";
+
+    private static final String ELASTICSEARCH_EMPTY_RESPONSE = "{\"total\": 0,\"cases\": []}";
+
+    @BeforeEach
+    void before() throws Exception {
+        WireMock.reset();
+        givenThat(get(GET_CASE_URL).willReturn(aResponse().withBody(MOCK_RESPONSE)));
+        given(messageReceiver.receive()).willReturn(MOCK_MESSAGE).willReturn(null);
+    }
 
     @DisplayName("Should call ccd to attach supplementary evidence for caseworker")
     @Test
     void should_call_ccd_to_attach_supplementary_evidence_for_caseworker() throws Exception {
+        // when
+        envelopeEventProcessor.processNextMessage();
+
+        // then
+        await()
+            .atMost(60, TimeUnit.SECONDS)
+            .pollInterval(2, TimeUnit.SECONDS)
+            .ignoreExceptions()
+            .until(() -> {
+                WireMock.verify(postRequestedFor(urlPathEqualTo(Environment.CASE_EVENT_URL)));
+
+                return true;
```

Comment:
will `.until` try again if `verify` fails? Does `verify` throw an exception?

Reply:
Yes it tries every 2 seconds for max 2 minutes.
`.atMost(60, TimeUnit.SECONDS)
            .pollInterval(2, TimeUnit.SECONDS)
`

Reply:
I mean, will the test not fail immediately due to `WireMock.verify(postRequestedFor(urlPathEqualTo(Environment.CASE_EVENT_URL)));` throwing an exception? does it handle code throwing exceptions or just predicate (true/false)

Reply:
It will fail if the verification fails

Reply:
so it will not try again if first try fails?

Reply:
discussed in person